src/adapter/src/coord/sequencer/inner/create_materialized_view.rs

```diff
@@ -577,6 +578,23 @@ impl Coordinator {
             global_lir_plan,
             ..
         } = stage;
+
+        // Validate the replacement target, if one is given.
+        // TODO(alter-mv): Could we do this already in planning?
+        if let Some(target_id) = replacement_target {
+            let Some(target) = self.catalog().get_entry(&target_id).materialized_view() else {
+                return Err(AdapterError::internal(
+                    "create materialized view",
+                    "replacement target not a materialized view",
+                ));
+            };
+
+            // For now, we don't support schema evolution for materialized views.
+            if &target.desc.latest() != global_lir_plan.desc() {
+                return Err(AdapterError::Unstructured(anyhow!("incompatible schemas")));
```

Comment:
Might eventually want a structured error for this one

Reply:
I think I left that here because I wanted to look into whether this check should/can be moved to planning instead, as a follow-up.